CMake/macros.cmake

```diff
@@ -141,6 +141,26 @@ FUNCTION( RUN_QMC_APP TESTNAME SRC_DIR PROCS THREADS TEST_ADDED TEST_LABELS ${AR
 ENDFUNCTION()
 
 
+# this disables the extremely slow setup of full system run tests
+# these prevent a reasonable compile debug loop if CMakeLists.txt files are touched since they
+# far exceed even the feature and compiler checks for time consumed when rerunning cmake.
+#
+# On leconte with rerun of cmake with
+# slow tests 27 secs to run cmake
+# with them disable 8 secs
+
+SET( QMC_DISABLE_SLOW_TESTS FALSE CACHE BOOL "Disable the slow cmake system test additions")
```

Comment:
1. please move this line to the CTEST section in the top level cmakelist.txt. Keep only functions in this macro.cmake.
2. We use "integration test" instead of "system test" in our documentation and manual. Please align the vocabulary.
3. "slow tests" sounds like the tests are slow to run but here is for tests slow to configure. So better call it QMC_DISABLE_TESTS_CONFIGURED_SLOWLY. Please leave a note before this line saying this will also disable part of deterministic tests and most users should not set this flag specifically.

Reply:
Maybe we should update the manual.

Integration tests are usually done without the entire system up, and between a well defined subset of object/modules. QMCPACK has a good bit "accidental" integration testing in the unit tests but very little that was written on purpose.
https://en.wikipedia.org/wiki/Integration_testing

It's hard to be precise but I wouldn't generally consider anything feeding input files to the main executable an integration test.